compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/converter/LightTreeRawFirExpressionBuilder.kt

```diff
@@ -498,7 +498,7 @@ class LightTreeRawFirExpressionBuilder(
             }
         }
 
-        val result = firExpression ?: buildErrorExpression(null, ConeNotAnnotationContainer("???"))
+        val result = firExpression ?: buildErrorExpression(annotatedExpression.toFirSourceElement(), ConeNotAnnotationContainer("???"))
```

Comment:
This must have a location, otherwise we get a crash in `FirContractChecker`.

```kotlin
object FirContractChecker {
    ...
    private fun checkUnresolvedEffects(
        ...
    ) {
        ...
            reporter.reportOn(unresolvedEffect.source, ...)
                                           // ^ source is asserted to be non-null inside reportOn()
        ...
    }
    ...
}
```

Reply:
Yep, this is a correct fix. Could you please check that the corresponding place in `PsiRawFirBuilder` also correctly sets the source?

Reply:
Yes, it sets it. In fact I checked that first and that was my leading clue that this one should be set too.

```kotlin
val result = rawResult as? FirAnnotationContainer
    ?: buildErrorExpression(
        expression.toFirSourceElement(),
        ConeNotAnnotationContainer(rawResult?.render() ?: "???")
    )
```